docs/spec/spec.md

```diff
@@ -0,0 +1,316 @@
+
+- [Source](#kind-source)
+- [Channel](#kind-channel)
+- [Subscription](#kind-subscription)
+- [Provider](#kind-provisioner)
+
+---
+
+## kind: Source
+
+### group: eventing.knative.dev/v1alpha1
+
+_Describes a specific configuration (credentials, etc) of a source system which
+can be used to supply events. A common pattern is for Sources to emit events to
+Channel to allow event delivery to be fanned-out within the cluster. They
+cannot receive events._
+
+### Object Schema
+
+#### Spec
+
+| Field         | Type                               | Description                                                                                                            | Limitations                                               |
+| ------------- | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------- |
+| provisioner\* | ProvisionerReference               | The provisioner used to create any backing resources and configuration.                                                | Immutable.                                                |
+| arguments     | runtime.RawExtension (JSON object) | Arguments passed to the provisioner for this specific source.                                                          | Arguments must validate against provisioner's parameters. |
+| channel       | ObjectRef                          | Specify an existing channel to use to emit events. If empty, create a new Channel using the cluster/namespace default. | Source will not emit events until channel exists.         |
+
+\*: Required
+
+#### Status
+
+| Field        | Type                      | Description                                                                                  | Limitations                                              |
+| ------------ | ------------------------- | -------------------------------------------------------------------------------------------- | -------------------------------------------------------- |
+| subscribable | Subscribable              | Pointer to a channel which can be subscribed to in order to receive events from this source. |                                                          |
+| provisioned  | []ProvisionedObjectStatus | Creation status of each Channel and errors therein.                                          | It is expected that a Source list all produced Channels. |
+| conditions   | Conditions                | Source conditions.                                                                           |                                                          |
+
+##### Conditions
+
+- **Ready.** True when the Source is provisioned and ready to emit events.
+- **Provisioned.** True when the Source has been provisioned by a controller.
+
+#### Events
+
+- Provisioned - describes each resource that is provisioned.
```

Comment:
@n3wscott should a source emit more events, than just `Provisioned` ? E.g. `Deprovisioned` or something that reflects a failed state ? 

(we are not talking here about the event that the source submits to a channel, right?)